policyengine_uk/variables/gov/dfe/extended_childcare_entitlement/conditions/extended_childcare_entitlement_work_condition.py

```diff
@@ -0,0 +1,54 @@
+from policyengine_uk.model_api import *
+
+
+class extended_childcare_entitlement_work_condition(Variable):
+    value_type = bool
+    entity = Person
+    label = "Work conditions for extended childcare entitlement"
+    documentation = "Whether the person/couple meets work requirements for extended childcare entitlement"
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        benunit = person.benunit
+        is_child = person("is_child", period)
+        in_work = person("in_work", period)
+        is_parent = person("is_parent", period)
+
+        # Get disability status
+        eligible_based_on_disability = (
+            add(
+                person,
+                period,
+                parameters(
```

Comment:
This code structure is hard to read- please get the right nodes in separate steps. Also, you should generally never call parameters(period) twice in a formula.